src/dropdown.stories.ts

```diff
@@ -129,18 +129,48 @@ const meta: Meta = {
         type: {
           summary: 'method',
           detail: `
-async (query: string): Promise<DropdownOption[] | void> => {
+// By default, \`filter()\` is implemented similar to the first example below. It filters against
+// Dropdown Options already present in Dropdown's default slot.
+//
+// You can override \`filter()\` to change what gets filtered out by returning the options
+// you want visible. The rest will be hidden:
+
+const dropdown = document.querySelector('glide-core-dropdown');
+
+dropdown.filter = async (query: string): Promise<DropdownOption[] | void> => {
   const options = [...this.querySelectorAll('glide-core-dropdown-option)];
 
   return options.filter(({ label }) => {
     return label.toLowerCase().includes(query.toLowerCase().trim());
   });
 }
 
-// When overriding this method, return the options you want visible. The rest will be hidden.
-//
-// If you fetch when filtering, this is the place to do it. Just make sure you've updated
-// Dropdown's default slot with the new set of options before you query for and filter them.
+// Alternatively, you can override \`filter()\` to support server-side filtering:
+
+class Component extends LitElement {
+  @state()
+  options = [{ label: 'One', key: 'one' }, { label: 'Two', key: 'two' }];
+
+  firstUpdated() {
+    this.#dropdownRef.value.filter = async (query: string): Promise<DropdownOption[] | void> => {
+      this.options = window.fetch(query);
```

Comment:
By no means a practical example. But I also didn't want to obscure the gist of it. Plus, whatever the example, consumers will have to tweak it.